src/core_plugins/vega/public/vega_editor.controller.js

```diff
@@ -0,0 +1,72 @@
+import { uiModules } from 'ui/modules';
+import hjson from 'hjson';
+import compactStringify from 'json-stringify-pretty-compact';
+
+const module = uiModules.get('kibana/vega', ['kibana']);
+module.controller('VegaEditorController', ($scope /*, kbnUiAceKeyboardModeService*/) => {
+
+  return new (class VegaEditorController {
+    constructor() {
+      $scope.aceLoaded = (editor) => {
+        editor.$blockScrolling = Infinity;
+
+        const session = editor.getSession();
+        session.setTabSize(2);
+        session.setUseSoftTabs(true);
+
+        // FIXME: enabling this service breaks ACE text wrapping on width resize
+        // kbnUiAceKeyboardModeService.initialize($scope, editor);
+
+        this.aceEditor = editor;
+      };
+
+      $scope.formatJson = (event) => {
+        this._format(event, compactStringify, {
+          maxLength: this._getCodeWidth(),
+        });
+      };
+
+      $scope.formatHJson = (event) => {
+        this._format(event, hjson.stringify, {
+          condense: this._getCodeWidth(),
+          bracesSameLine: true,
+          keepWsc: true,
+        });
+      };
+
+      // $hack: by default, Vega vis should enable auto-apply, as it is a more natural way
+      // to develop Vega visualizations, and consistent with other Vega tools and Kibana Vega plugin.
+      $scope.$parent.$parent.$parent.$parent.autoApplyEnabled = true;
```

Comment:
This should be specified in your `visType.editorConfig` like https://github.com/elastic/kibana/blob/master/src/core_plugins/markdown_vis/public/markdown_vis.js#L37

Reply:
Yep, the `enableAutoApply: true` is already there - this code was meant to be a joke for Peter and to convince him to add the `enableAutoApply` :)

Reply:
lol 🤢